spec/features/checkouts_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Checkouts', type: :feature do
+  # patron2 below has 2 checkouts
+  let(:mock_user) { 'patron2' }
+
+  before do
+    login_as username: 'PATRON2', patron_key: mock_user
+  end
+
+  after do
+    Redis.new.del 'item_type_map'
+  end
+
+  context 'when patron has checkouts to display', js: true do
+    before do
+      visit checkouts_path
+      login_as username: 'PATRON2', patron_key: mock_user
+    end
+
+    it 'displays total checked out correctly' do
+      expect(page).to have_content 'Total Checked Out: 2'
+    end
+
+    it 'displays total # of recalled items correctly' do
+      expect(page).to have_content 'Recalled: 1'
+    end
+
+    it 'displays total # of overdue items correctly' do
+      expect(page).to have_content 'Overdue: 2'
+    end
+
+    it 'displays checkout\'s item\'s title / author' do
+      expect(page).to have_content 'The labor situation in the Las Mercedes free'\
+                              ' trade zone in Managua, Nicaragua / Levit, Alex C.'
+    end
+
+    it 'displays call number correctly' do
+      expect(page).to have_content 'HD8110.5.C667 2007'
+    end
+
+    it 'displays checkout\'s item\'s canonical item link' do
+      expect(page).to have_link('The politics of labor reform in Latin America : between flexibility and rights'\
+                    ' / Cook, Maria Lorena.', href: 'https://catalog.libraries.psu.edu/catalog/3591032')
+    end
+
+    it 'displays checkout\'s item\'s item type' do
+      expect(page).to have_content 'Penn State Thesis (Bachelors)'
+    end
+
+    it 'displays checkout\'s renewal count' do
+      expect(page).to have_css('[id="checkout2145643:5:1"] .renewal_count', text: '69')
+    end
+
+    it 'displays checkout\'s due date correctly when not recalled' do
+      expect(page).to have_css('[id="checkout2145643:5:1"] .due_date', text: 'May 7, 2020')
+    end
+
+    it 'displays checkout\'s estimated overdue amount when item is overdue' do
+      expect(page).to have_content '$10.00'
+    end
+
+    it 'displays checkout\'s due date correctly when recalled' do
+      expect(page).to have_css('[id="checkout3591032:1:1"] .due_date',
+                               text: 'Recalled<br>August 4, 2020 10:34am<br>May 7, 2020')
+    end
+
+    it 'displays checkout\'s status correctly when overdue' do
+      expect(page).to have_css('[id="checkout2145643:5:1"] .status', text: 'Overdue')
+    end
+
+    it 'displays checkout\'s status correctly when claims returned' do
+      expect(page).to have_css('[id="checkout3591032:1:1"] .status', text: 'Claims Returned')
+    end
+
+    it 'displays a renew all select' do
+      expect(page).to have_unchecked_field 'renewal_list__2145643:5:1'
+    end
+
+    it 'displays a renew button' do
+      expect(page).to have_button 'Renew'
+    end
```

Comment:
Part of me wonders if we need all these (or any of these really). Or at least we should think about combining some or all of them because of the test overhead maybe. I'm just not sure what all they provide in terms of assurance in general, and now they're more expensive. 🤔

Reply:
These were in the view spec, I moved them over here because I did not how to test them within a view spec anymore with asyncing work changing thigs 🤔

Reply:
I think I figure out the view spec, I will move them back to a view spec.

Reply:
I know why you moved them here, no I get that. But again, do we need them? What do they actually give us? Take them out and run rake and see what overhead they cost, you know?

Reply:
moved them back to a view spec, and cut a few out, there are couple things like displaying Total Overdue, Recalled etc. numbers etc. that I think not covered by other tests. It seems to be not taking much time to run but let me know if you think if these are still an overhead? 

``` 
bundle exec rspec spec/views/checkouts/all.html.erb_spec.rb
...........

Finished in 0.15921 seconds (files took 2.39 seconds to load)
11 examples, 0 failures
```

Reply:
much much better. Running all these in a feature spec with `js` on was costing a lot.